opentok-angular.js

```diff
@@ -311,3 +293,94 @@ ng.module('opentok', [])
       };
     }
   ]);
+
+  const captionSubscriberTracker = (OTSession, $rootScope) => {
+    const MAX_SUBS_ON_SCREEN = 4;
+    const CAPTIONS_TIMEOUT = 5 * 1000;
+
+    const captionBox = document.getElementById('caption-render-box');
+
+    let namesByConnectionId = {};
+
+    OTSession.session.on('signal:name', (event) => {
+      namesByConnectionId[event.from.connectionId] = event.data;
+    });
+    const getNameFromConnection = (connection) => {
+      let id = connection.creationTime.toString();
+      id = id.substring(id.length - 6, id.length - 1);
+      return `Guest${id}`;
+    };
+    const getName = (from) => {
+      if (!namesByConnectionId[from.connectionId]) {
+        namesByConnectionId[from.connectionId] = getNameFromConnection(from);
+      }
+      return namesByConnectionId[from.connectionId];
+    };
+
+    // we should remove the last element of the array if it exceeds the size
+    const captionsArray = $rootScope.captionsArray;
+    // the object should have the shape {caption, streamId, timeout, name}
```

Comment:
Nit: this comment is superfluous.  Its shape can be inferred in the code below.

Reply:
Yeah, I think a lot of the comments in here for scaffolding for when I was writing the code and are now superfluous. I will clean them up.